hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -798,7 +795,9 @@ protected void archive(HoodieTable table) {
     }
     try {
       final Timer.Context timerContext = metrics.getArchiveCtx();
-      // We cannot have unbounded commit files. Archive commits if we have to archive
+      // We cannot have unbounded commit files. Archive commits if we have to archive.
+      // Reload table timeline to reflect the latest commit.
+      table.getMetaClient().reloadActiveTimeline();
```

Comment:
If we reload timeline here, it is possible that write client and table service client have different fs view. Is that expected? Also, why should we not then reload timeline for other table services before executing that table service?

Reply:
Only the archive needs this now because the cleaning may be executed before the archiving. In other other cases, there is no need to refresh either the timeline or the fs view.